### PR TITLE
feat: query analysis (explain) feature

### PR DESCRIPTION
**Pull request by saravmajestic**
## Overview
Query explanation feature: https://www.notion.so/altimate/Query-Explanation-in-DataPilot-a49664edab33452fb95de1fc2e853d7b

### Changes made:
- moved `Explain selected SQL (Preview feature)` from editor context menu to datapilot sub menu
- converted existing explain button in actions and query panel to use datapilot new explain feature
- implemented datapilot chat flow for query explain with keybindings 
- added help section in datapilot
- added storybook for query explain variations
- added max followup limit
- render ai response as markdown

Demo: https://www.awesomescreenshot.com/video/24896369?key=a30be8cd524887795b74d4cff0f1d91f

## Checklist

- [x] I have run this code and it appears to resolve the stated issue
- [x] `README.md` updated and added information about my change

### CODE REVIEW

File: src/commands/runModel.ts [hunk 1]
```diff
@@ -78,14 +78,6 @@ export class RunModel {
     this.executeSQL(window.activeTextEditor!.document.uri, query, queryName);
   }
 
-  getSummaryOnActiveWindow() {
```
Comment (saravmajestic) [inline src/commands/runModel.ts:81]: Cleaning up existing query explain code

File: webview_panels/.storybook/__mocks__/vscode.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+let data = undefined;
+let timer = 0;
+let func = undefined;
+
+export const vscode = {
```
Comment (saravmajestic) [inline webview_panels/.storybook/__mocks__/vscode.ts:5]: This is for storybook integration helps in developing interactions in stories

File: webview_panels/src/testUtils/datapilot/queryAnalysis.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
+import { Sync, each } from "factory.ts";
+import { faker } from "@faker-js/faker";
+import { RequestState, RequestTypes } from "@modules/dataPilot/types";
+import {
+  DatapilotQueryAnalysisChat,
+  QueryAnalysisType,
+  QueryAnalysisResult,
+} from "@modules/dataPilot/components/queryAnalysis/types";
+
+export const DatapilotQueryAnalysisFactory =
```
Comment (saravmajestic) [inline webview_panels/src/testUtils/datapilot/queryAnalysis.ts:10]: test data for storybook

**Review by mdesmet — COMMENTED** [review]
LGTM, can you check the commands?

File: package.json [hunk 4]
```diff
@@ -271,6 +271,14 @@
           "dark": "./media/images/run-dark.svg"
         }
       },
+      {
+        "command": "dbtPowerUser.openDatapilotWithQuery",
+        "title": "Start Chat"
```
[the remark this answers is not in the document]
Comment (saravmajestic) [reply package.json:276]: Added `category` to show under `dbt Power User`.
Screenshot 2024-02-13 at 8 56 57 PM

**Pull request merged**